src/components/settings-address-book.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+import {AddressEmpty} from '@app/components/address-empty';
+import {AddressHeader} from '@app/components/address-header';
+import {ListContact} from '@app/components/list-contact';
+import {
+  Box,
+  Icon,
+  IconButton,
+  QRScanner,
+  Text,
+  TextField,
+} from '@app/components/ui';
+import {createTheme} from '@app/helpers';
+import {withActionsContactItem} from '@app/hocs';
+import {I18N, getText} from '@app/i18n';
+import {
+  LIGHT_GRAPHIC_BASE_2,
+  LIGHT_GRAPHIC_GREEN_1,
+  LIGHT_TEXT_BASE_1,
+} from '@app/variables';
+
+const ListOfContacts = withActionsContactItem(ListContact, {
+  nextScreen: 'settingsContactEdit',
+});
+
+type SettingsAddressBookProps = {
+  onPressAddress: () => void;
+  onPressAdd: () => void;
+  onPressClear: () => void;
+  onPressQR: () => void;
+  setSearch: (value: string) => void;
+  canAdd: boolean;
+  search: string;
+};
+
+export const SettingsAddressBook = ({
+  onPressAddress,
+  onPressAdd,
+  onPressClear,
+  onPressQR,
+  canAdd,
+  search,
+  setSearch,
+}: SettingsAddressBookProps) => {
+  return (
+    <>
+      <TextField
+        label={getText(I18N.settingsAddressBookLabel)}
+        style={styles.input}
+        placeholder={getText(I18N.settingsAddressBookPlaceholder)}
+        value={search}
+        onChangeText={setSearch}
+        multiline
+        autoFocus
+        rightAction={
+          search === '' ? (
+            <IconButton onPress={onPressQR}>
+              <QRScanner color={LIGHT_GRAPHIC_GREEN_1} style={styles.icon} />
```

Comment:
Replace this with an Icon component with the appropriate size
```jsx
<Icon i24 name="qr_scanner" color={Color.graphicGreen1} />
```

Reply:
done